lib/jira/client/axios.js

```diff
@@ -1,86 +1,61 @@
 const axios = require('axios')
 const jwt = require('atlassian-jwt')
+const querystring = require('querystring')
 const url = require('url')
 
 const { logger } = require('probot/lib/logger')
 
 const instance = process.env.INSTANCE_NAME
 const iss = 'com.github.integration' + (instance ? `.${instance}` : '')
 
+/*
+ * Create an Atlassian request from an Axios config.
+ * Compatible with jwt.createQueryStringHash.
+ */
+jwt.reqFromAxiosConfig = (config) => {
+  const parsedUrl = url.parse(config.url)
+  const parsedQuery = querystring.parse(parsedUrl.query)
+
+  return {
+    method: config.method,
+    originalUrl: parsedUrl.pathname,
+    query: Object.assign(parsedQuery, config.params)
+  }
+}
+
 function getAuthMiddleware (secret) {
   return (config) => {
-    const { query, pathname } = url.parse(config.url, true)
-
-    const jwtToken = jwt.encode({
-      ...getExpirationInSeconds(),
-      iss,
-      qsh: jwt.createQueryStringHash({
-        method: config.method,
-        originalUrl: pathname,
-        query
-      })
-    }, secret)
-
-    return {
-      ...config,
-      headers: {
-        ...config.headers,
-        Authorization: `JWT ${jwtToken}`
-      }
-    }
+    const qsh = jwt.createQueryStringHash(jwt.reqFromAxiosConfig(config))
+    const jwtToken = jwt.encode({ ...getExpirationInSeconds(), iss, qsh }, secret)
+
+    config.headers['Authorization'] = `JWT ${jwtToken}`
+
+    return config
   }
 }
 
 function getErrorMiddleware (id, installationId) {
   return (error) => {
-    const { status, statusText } = error.response || {}
-    logger.debug({
-      id,
-      installation: installationId,
-      params: error.config.fields
-    }, `Jira request: ${error.request.method} ${error.request.path} - ${status} ${statusText}`)
+    if (error.response) {
+      logResponse(id, installationId, error.response)
+    }
 
     return Promise.reject(error)
   }
 }
 
 function getSuccessMiddleware (id, installationId) {
   return (response) => {
-    logger.debug({
-      id,
-      installation: installationId,
-      params: response.config.fields
-    }, `Jira request: ${response.config.method.toUpperCase()} ${response.config.originalUrl} - ${response.status} ${response.statusText}`)
-
+    logResponse(id, installationId, response)
     return response
   }
 }
 
-function getUrlMiddleware () {
-  return (config) => {
-    let { query, pathname, ...rest } = url.parse(config.url, true)
-    config.fields = config.fields || {}
-
-    for (let field in config.fields) {
-      if (pathname.includes(`:${field}`)) {
-        pathname = pathname.replace(`:${field}`, config.fields[field])
-      } else {
-        query[field] = config.fields[field]
-      }
-    }
-
-    config.fields['baseUrl'] = config.baseURL
+const logResponse = function (id, installationId, response) {
+  const { status, statusText } = response
+  const { method, path } = response.request
 
-    return {
-      ...config,
-      originalUrl: config.url,
```

Comment:
Transferring @scepticulous's comment from the last PR about why we're removing this; Looks like `UrlMiddleware` was adding an `originalUrl` and `url` field to the request config. Do we not need these anymore with the axios upgrade?